tests/integration/test_backups_ceph.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+import dataclasses
+import json
+import logging
+import os
+import socket
+import subprocess
+import time
+
+import boto3
+import botocore.exceptions
+import pytest
+from pytest_operator.plugin import OpsTest
+
+from . import architecture, markers
+from .helpers import (
+    backup_operations,
+)
+from .juju_ import juju_major_version
+
+logger = logging.getLogger(__name__)
+
+S3_INTEGRATOR_APP_NAME = "s3-integrator"
+if juju_major_version < 3:
+    tls_certificates_app_name = "tls-certificates-operator"
+    if architecture.architecture == "arm64":
+        tls_channel = "legacy/edge"
+    else:
+        tls_channel = "legacy/stable"
+    tls_config = {"generate-self-signed-certificates": "true", "ca-common-name": "Test CA"}
+else:
+    tls_certificates_app_name = "self-signed-certificates"
+    if architecture.architecture == "arm64":
+        tls_channel = "latest/edge"
+    else:
+        tls_channel = "latest/stable"
+    tls_config = {"ca-common-name": "Test CA"}
+
+backup_id, value_before_backup, value_after_backup = "", None, None
+
+
+@dataclasses.dataclass(frozen=True)
+class ConnectionInformation:
+    access_key_id: str
+    secret_access_key: str
+    bucket: str
+
+
+@pytest.fixture(scope="session")
+def microceph():
+    if not os.environ.get("CI") == "true":
+        raise Exception("Not running on CI. Skipping microceph installation")
+    logger.info("Setting up TLS certificates")
+    subprocess.run(["sudo", "openssl", "genrsa", "-out", "./ca.key", "2048"], check=True)
+    subprocess.run(
+        [
+            "sudo",
```

Comment:
I don't think you need root to run the openssl commands.

Reply:
You are right, removing it did not make a difference. Thanks for pointing it out!